packages/core/test/leakDetection.ts

```diff
@@ -0,0 +1,53 @@
+import { display } from '../src/tools/display'
+import { getCurrentJasmineSpec } from './getCurrentJasmineSpec'
+
+let originalAddEventListener: typeof EventTarget.prototype.addEventListener
+let originalRemoveEventListener: typeof EventTarget.prototype.removeEventListener
+let wrappedListeners: {
+  [key: string]: Map<EventListenerOrEventListenerObject | null, EventListenerOrEventListenerObject | null>
+}
+
+export function startLeakDetection() {
+  wrappedListeners = {}
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  originalAddEventListener = EventTarget.prototype.addEventListener
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  originalRemoveEventListener = EventTarget.prototype.removeEventListener
+
+  EventTarget.prototype.addEventListener = function (event, listener, options) {
+    if (!wrappedListeners[event]) {
+      wrappedListeners[event] = new Map()
+    }
+    const wrappedListener = withLeakDetection(event, listener as EventListener)
+    wrappedListeners[event].set(listener, wrappedListener)
+    return originalAddEventListener.call(this, event, wrappedListener, options)
+  }
+  EventTarget.prototype.removeEventListener = function (event, listener, options) {
+    const wrappedListener = wrappedListeners[event]?.get(listener)
+    wrappedListeners[event]?.delete(listener)
+    return originalRemoveEventListener.call(this, event, wrappedListener || listener, options)
+  }
+}
+
+export function stopLeakDetection() {
+  EventTarget.prototype.addEventListener = originalAddEventListener
+  EventTarget.prototype.removeEventListener = originalRemoveEventListener
+  Object.keys(wrappedListeners).forEach((key) =>
+    wrappedListeners[key].forEach((_, listener) => wrappedListeners[key].delete(listener))
+  )
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏it could be simpler with:

```suggestion
  Object.values(wrappedListeners).forEach((wrappedListenersForEvent) =>
    wrappedListenersForEvent.clear()
  )
```

or even simpler:

```suggestion
wrappedListeners = {}
```